examples/response_stream/server/routes/redux_stream.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IRouter, Logger } from '@kbn/core/server';
+import { streamFactory } from '@kbn/ml-response-stream/server';
+
+import {
+  updateProgress,
+  addToEntity,
+  deleteEntity,
+  error,
+  type ReduxStreamApiAction,
+} from '../../common/api/redux_stream/data_slice';
+import { reducerStreamRequestBodySchema } from '../../common/api/reducer_stream';
+import { RESPONSE_STREAM_API_ENDPOINT } from '../../common/api';
+
+import { entities, getActions } from './shared';
+
+export const defineReduxStreamRoute = (router: IRouter, logger: Logger) => {
+  router.versioned
+    .post({
+      path: RESPONSE_STREAM_API_ENDPOINT.REDUX_STREAM,
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: reducerStreamRequestBodySchema,
+          },
+        },
+      },
+      async (context, request, response) => {
+        const maxTimeoutMs = request.body.timeout ?? 250;
+        const simulateError = request.body.simulateErrors ?? false;
+
+        let logMessageCounter = 1;
+
+        function logDebugMessage(msg: string) {
+          logger.debug(`Response Stream Example #${logMessageCounter}: ${msg}`);
+          logMessageCounter++;
+        }
+
+        logDebugMessage('Starting stream.');
+
+        let shouldStop = false;
+        request.events.aborted$.subscribe(() => {
+          logDebugMessage('aborted$ subscription trigger.');
+          shouldStop = true;
+        });
+        request.events.completed$.subscribe(() => {
+          logDebugMessage('completed$ subscription trigger.');
+          shouldStop = true;
+        });
+
+        const { end, push, responseWithHeaders } = streamFactory<ReduxStreamApiAction>(
+          request.headers,
+          logger,
+          request.body.compressResponse,
+          request.body.flushFix
+        );
+
+        const actions = getActions(simulateError);
+
+        let progress = 0;
+
+        async function pushStreamUpdate() {
+          await new Promise((resolve) =>
+            setTimeout(resolve, Math.floor(Math.random() * maxTimeoutMs))
+          );
+          try {
+            progress++;
+
+            if (progress > 100 || shouldStop) {
+              end();
+              return;
+            }
+
+            push(updateProgress(progress));
+
+            const randomEntity = entities[Math.floor(Math.random() * entities.length)];
+            const randomAction = actions[Math.floor(Math.random() * actions.length)];
+
+            if (randomAction === 'add') {
```

Comment:
super duper nit: I think a switch statement would be more readable here

Reply:
updated in [737bdfd](https://github.com/elastic/kibana/pull/182690/commits/737bdfdc9bff4aa741c44d7b763d52ea006eb4ed).